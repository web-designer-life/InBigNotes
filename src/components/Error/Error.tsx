import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../constants';
import { 
    Container, 
    Message, 
    BackLink,
} from './style';

export default class Error extends React.Component {
	render() {
        return (
            <Container>
                <Message>404 - Not Found!</Message>
                <Link to={ROUTES.HOME}>
                    <BackLink>Go home</BackLink>
                </Link>
            </Container>
        );
    };
};
