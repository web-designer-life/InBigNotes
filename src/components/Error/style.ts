import styled from 'styled-components';

const Container = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100vh;
	flex-direction: column;
`;

const Message = styled.h1`
    margin: 10px;
`;

export {
    Container,
    Message,
}