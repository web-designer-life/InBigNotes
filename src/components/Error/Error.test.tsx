import React from 'react'
import Error from '../Error/Error';
import { render } from 'enzyme';

describe("Error component", () => {
    describe("Has props", () => {
        it("should render Post component", () => {
            const component = render(<Error />);
            expect(component).toMatchSnapshot();
        });
    });
  
    describe("defaultProps", () => {
        it("should use default navigateToPage", () => {
            const result = Error.defaultProps.navigateToPage();
            expect(result).toBe(undefined);
        });
    });
});
