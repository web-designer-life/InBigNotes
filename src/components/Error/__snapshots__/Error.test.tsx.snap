// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component Has props should render Post component 1`] = `
<div
  class="sc-gsTEea kOLVVs"
>
  <h1
    class="sc-dlfnuX fsWPCe"
  >
    404 - Not Found!
  </h1>
  <button
    class="sc-bdfBQB leOgNx"
    color="green"
    type="button"
  >
    Go home
  </button>
</div>
`;
